pkg/engine/engine.go

```diff
@@ -207,7 +212,7 @@ func (e *Engine) Start(ctx Context, input string) (*Return, error) {
 		InternalSystemPrompt: tool.Parameters.InternalPrompt,
 	}
 
-	for _, subToolName := range tool.Parameters.Tools {
+	for _, subToolName := range append(tool.Parameters.Tools, tool.Parameters.Credentials...) {
```

Comment:
Why do we give the credential tools to the LLM?

Reply:
Fixed